client/blocks/reader-post-card/index.jsx

```diff
@@ -158,15 +161,33 @@ export default class ReaderPostCard extends React.Component {
 				</StandardPost>;
 		}
 
+		// set up post byline
+		let postByline;
+
+		if ( isDiscoverStream && ! isEmpty( discoverPick ) ) {
+			// create a post like object with some props from the discover post
```

Comment:
This seems like it should be a post-normalizer rule: `rule-post-to-open`.

Then we probably wouldn't need any special cases in this file!